cdap-ui/app/cdap/components/PipelineConfigurations/index.js

```diff
@@ -126,9 +126,16 @@ export default class PipelineConfigurations extends Component {
   }
 
   render() {
-    let tabConfig = TabConfig;
-    if (!this.props.isBatch) {
-      tabConfig.tabs = tabConfig.tabs.slice(0, -1);
+    let tabConfig;
+    if (this.props.isBatch) {
+      tabConfig = TabConfig;
+    } else {
+      tabConfig = {...TabConfig};
```

Comment:
I think it's because I'm reassigning the `tabs` property in `tabConfig` instead of modifying it directly, so that's why it doesn't mutate the `tabs` property of the original object